hudi-common/src/main/java/org/apache/hudi/metadata/BaseTableMetadata.java

```diff
@@ -418,6 +417,8 @@ private void checkForSpuriousDeletes(HoodieMetadataPayload metadataPayload, Stri
 
   protected abstract Map<String, HoodieRecord<HoodieMetadataPayload>> getRecordsByKeys(List<String> keys, String partitionName);
 
+  protected abstract Map<String, List<HoodieRecord<HoodieMetadataPayload>>> getAllRecordsByKeys(List<String> keys, String partitionName);
```

Comment:
@nsivabalan I added a separate method (yet to fully implement) for separation as we will have duplicates only in case of RLI for now. But, should we chnage the return type existing `getRecordsByKeys`?